driven/storage/adapter.go

```diff
@@ -909,8 +909,40 @@ func (sa *Adapter) FindGroupByTitle(clientID string, title string) (*model.Group
 }
 
 // FindGroups finds groups
-func (sa *Adapter) FindGroups(clientID string, category *string, privacy *string, title *string, offset *int64, limit *int64, order *string) ([]model.Group, error) {
+func (sa *Adapter) FindGroups(clientID string, userID *string, category *string, privacy *string, title *string, offset *int64, limit *int64, order *string) ([]model.Group, error) {
 	filter := bson.D{primitive.E{Key: "client_id", Value: clientID}}
+
+	if userID != nil {
+		innerOrFilter := bson.A{
+			bson.D{
+				{"members",
+					bson.D{
+						{"$elemMatch",
+							bson.D{
+								{"user_id", *userID},
+								{"$or",
+									bson.A{
+										bson.D{{"status", "admin"}},
+										bson.D{{"status", "member"}},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			bson.D{{"privacy", bson.D{{"$ne", "private"}}}},
+		}
+
+		if title != nil {
```

Comment:
Do we also need to account for the `hidden_for_search` field here?

Reply:
Nice catch! Thank you!